src/js/utils/date-utils.js

```diff
@@ -0,0 +1,283 @@
+         * @param {object} localTime A moment.js datetime object offset to a determined timezone.
+         * @param {string} language An ISO 639-1 language string (e.g. 'en', 'fr').
+         * @param {string} format (optional) A desired format constant.
+         * @return {string} A language/timezone aware localized string
+        */
+        timeDisplay = function(localTime, language, format) {
+            var displayLang = 'en-US';
+            var displayFormat = DATE_FORMATS.DEFAULT_DATE_TIME_FORMAT;
+            var displayTime;
+            if (isValid(language)) {
+                displayLang = language;
+            } else {
+                displayLang = window.navigator.userLanguage || window.navigator.language;
+            }
+
+            // Determine the desired format, default to DEFAULT_DATE_TIME_FORMAT
+            if (isValid(format) && format in DATE_FORMATS) {
+                displayFormat = DATE_FORMATS[format];
+            }
+            displayTime = localTime.locale(displayLang).format(displayFormat);
+            // Attempt to repair to UTC/US
+            if (isValid(displayTime) === false) {
+                displayTime = localTime.tz('UTC').locale('en').format(displayFormat);
+            }
+            if (isValid(displayTime) === false) {
+                displayTime = '';
+            }
+            return displayTime;
+        };
+
+        /**
+         *
+         * This will attempt render a localized date/time in a
+         * standard format.
+         *
+         * This should take a context dict, with an additional (and optional)
+         * handler for a modifier_string (e.g. 'due', 'deadline:', etc.) via
+         * StringUtils.
+         *
+         * Please Note: the preferred method is via underscore in the mako
+         * templates, however this will also work.
+         *
+         * @param {Object} context
+         * @returns {string} A formatted, timezone-offset, internationalized date and time.
+         *
+         * Example Context:
+         *
+         *~~~ javascript
+         * var context = {
+         *    datetime: '2016-10-14 08:00:00+00:00',
+         *    timezone: 'Pacific/Kiritimati',
+         *    language: 'ar',
+         *    format: 'DEFAULT_LONG_DATE_FORMAT'
+         *    };
+         * return_string = DateUtils.localize(context);
+         *~~~
+         *
+         * - datetime: {string} The (UTC) datetime server-side string.
+         * - modifier: {string} (optional) 'due', 'deadline:', etc..
+         * - timezone: {string} (optional) the user set time zone preference, or 'None' if unset.
+         * This should be in a format like 'Region/Locale', e.g. 'Atlantic/Azores'. Defaults to 'UTC'.
+         * - language: {string} (optional), the user set language, an ISO 639-1 language string, defaults to 'en'
+         * - format: {string} (optional), a format constant. Defaults to 'DEFAULT_DATE_TIME_FORMAT'.
+         */
+        localize = function(context) {
```

Comment:
@nasthagiri I was under the impression that, stylistically, over, like three arguments should be passed in as a single argument. Should this be named something else?